examples/vault-iam-auth/main.tf

```diff
@@ -0,0 +1,285 @@
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY A VAULT SERVER CLUSTER AND A CONSUL SERVER CLUSTER IN AWS
+# This is an example of how to launch a vault cluster and then authenticate an instance to the cluster
+# ---------------------------------------------------------------------------------------------------------------------
+
+terraform {
+  required_version = ">= 0.11.0"
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# INSTANCE THAT WILL AUTHENTICATE TO VAULT USING IAM METHOD
+# ---------------------------------------------------------------------------------------------------------------------
+resource "aws_instance" "example_auth_to_vault" {
+  ami           = "${var.ami_id}"
+  instance_type = "t2.micro"
+  subnet_id     = "${data.aws_subnet_ids.default.ids[0]}"
+  key_name      = "${var.ssh_key_name}"
+
+  # Security group that opens the necessary ports for consul
+  # And security group that opens the port to our simple web server
+  security_groups = [
+    "${module.consul_cluster.security_group_id}",
+    "${aws_security_group.auth_instance.id}",
+  ]
+
+  user_data            = "${data.template_file.user_data_auth_client.rendered}"
+  iam_instance_profile = "${aws_iam_instance_profile.example_instance_profile.name}"
+
+  tags {
+    Name = "${var.auth_server_name}"
+  }
+
+  # These two blocks below exist to provision a script that uses an AWS SDK to sign a request
+  provisioner "file" {
+    source      = "${path.module}/auth-signature-scripts"
+    destination = "/tmp"
+  }
```

Comment:
Curious why you use a provisioner here rather than packaging the python script into an AMI using Packer?

Reply:
I thought about that, but since this script is only used in this one example I decided to ship it like this. Should I ship it with packer even though it's for this one case?

Reply:
We should prob promote proper coding practices when we can. Of course, it's always a balance between ease of trying the example, security, and testability... But we already build AMIs for the Vault stuff, so we may as well add this script to that.

Reply:
Ok!